### ISSUE TITLE
`ERR_BLOCKED_BY_CLIENT` when clicking on links from email newsletters

### ISSUE THREAD

**Issue by HarryCutts** [EFForg/privacybadger#3000]
I've recently started getting `ERR_BLOCKED_BY_CLIENT` errors when I open links with some kind of tracking from Gmail messages. The full text of the error page is:

> **pkehgijcmpdhfbdbbnkijodmdjhbjlgp is blocked**
>
> This page has been blocked by Chrome
>
> ERR_BLOCKED_BY_CLIENT

![Screenshot 2024-07-21 12 58 08](https://github.com/user-attachments/assets/8242e9a3-ad11-4307-a632-587ff2cc0a29)

The URL bar for this error page shows a Privacy Badger URL. For example, when I click a link to this URL:

> https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E7t_p3GMT%2BhFqVLRh5g8VQaXBoi6YLCuqL_En1aSfupao2OO2ZLaytyq52_YMDerQiVpfyrTSRQRDfF44aHzn9_VIWlBTbSQye0EU

I get an error page with this URL:

> chrome-extension://pkehgijcmpdhfbdbbnkijodmdjhbjlgp/data/web_accessible_resources/redirect.html?url=https://email.triodos.co.uk/optiext/optiextension.dll?ID%3DE89E7t_p3GMT%252BhFqVLRh5g8VQaXBoi6YLCuqL_En1aSfupao2OO2ZLaytyq52_YMDerQiVpfyrTSRQRDfF44aHzn9_VIWlBTbSQye0EU

Notably, this is only a problem if I click on the link in the Gmail Web interface. If I right-click, choose "Copy link address", then paste it into the browser's address bar, it works fine. Similarly, if I use an email's "Open in web browser" option (where available) I can click links there.

Disabling Privacy Badger for Gmail or the domain that the link points to doesn't help. Nothing appears in the console for Privacy Badger's service worker when clicking on a problematic link.

Thanks for reading! Let me know if there are any other details I can provide. It seems GitHub doesn't let me attach .eml files for the emails, but I'm happy to forward them to someone if that would help.

## Versions

I'm running Google Chrome Dev 128.0.6600.0 (Official Build) dev (64-bit), on ChromeOS.

My Privacy Badger version is 2024.7.17.

## URLs

This only seems to affect links that have some kind of tracking going on, like you often find in email newsletters. Here are the ones I've encountered the issue with so far:

* All the ones in a bank newsletter:
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E7t_p3GMT%2BhFqVLRh5g8VQaXBoi6YLCuqL_En1aSfupao2OO2ZLaytyq52_YMDerQiVpfyrTSRQRDfF44aHzn9_VIWlBTbSQye0EU
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E7t_p3GMT%2BhFqVLRh5g8VQaXBoi6YLCuqL_En1aSfupao2OO2ZLaytyq52_YMDerQiVpfyPTSRQRDfF44aHzn9_VIWlBTaKdDT4Ee
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E6LJgw6WvPdGCqlX4W1WCsWCXOIB08G9T30znSq77n%2Bao2%2BQW8KbjABbpTmF%2Bk3F8ywPaRatVmxNtdl0uiluCeISumOiQJuJYOY_n
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E5O_rJSdyGNxeHBG129757fyg_E1lrxzZwd%2BgyzZcrd_o2EZ9u15zYKYxmblykIzdeLSwz1iUSsFRPKqMz6Byjl_u8e0eZHY5F5EO
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E6Mk%2Bcw2wJNcses8GTHnJCDVtcnwfLplBYQ_09SIVmUlo3_2XBORZjiwZmMZryxfTHh80nrchigs4pdX6_f3bFFRWDKCRnPFE4WEP
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E6c29kbG9z7F46TVpirE5pDEWtw7UnMBi7%2B7X_gcnGP4U3vn0ElgR3RgoHRjMg9t32weD7tHZWsM7T7I%2BLaKp2j9s%2BR_EbyJpWAE1
  * https://email.triodos.co.uk/optiext/optiextension.dll?ID=E89E8IcaCotzbJ66A1F_1zyhmTN_w1V3W1ebYQvcCv5Xw8UU2bDtrZ7WOVAw1jLlCm0FldidE0tXSKW78Ui2BvStxekW8TQyWdry8L_d
* All of the ones from a forwarded [event announcement](https://mailchi.mp/caa312245968/sir-philip-pullman-and-george-rr-martin-talk-fantasy-in-oxford?e=b589e196bf) that seems to use MailChimp and list-manage.com:
  * https://mailchi.mp/caa312245968/sir-philip-pullman-and-george-rr-martin-talk-fantasy-in-oxford?e=b589e196bf
  * https://ousnorcal.us8.list-manage.com/track/click?u=276af22e5f66fc1a24404759f&id=066cd9c6d1&e=b589e196bf
  * https://ousnorcal.us8.list-manage.com/track/click?u=276af22e5f66fc1a24404759f&id=b7b4d8d587&e=b589e196bf
  * https://ousnorcal.us8.list-manage.com/track/click?u=276af22e5f66fc1a24404759f&id=f080b05b13&e=b589e196bf
  * https://ousnorcal.us8.list-manage.com/track/click?u=276af22e5f66fc1a24404759f&id=03945640af&e=b589e196bf
* Some in an order confirmation email that uses Mandrill:
  * https://mandrillapp.com/track/click/30904755/www.blindsdirect.co.uk?p=eyJzIjoiT3RseXpnempjVWZWc2Y2TllVSHJwa3lxNjFvIiwidiI6MSwicCI6IntcInVcIjozMDkwNDc1NSxcInZcIjoxLFwidXJsXCI6XCJodHRwczpcXFwvXFxcL3d3dy5ibGluZHNkaXJlY3QuY28udWtcXFwvZGVsaXZlcnktaW5mb3JtYXRpb24_dXRtX21lZGl1bT1lbWFpbCZ1dG1fc291cmNlPXRyYW5zYWN0aW9uYWwmdXRtX2NhbXBhaWduPUJsaW5kcytEaXJlY3QrLStTYW1wbGVzKzErZGF5K2F1dG9tYXRpb25cIixcImlkXCI6XCI1NDk5MGNhNjIyMTk0OTc0OTc2NDI3ZGFkODZiM2ZjNlwiLFwidXJsX2lkc1wiOltcIjRjODMwNWMyYzBjZDU1MGFhMGYxMDM2OTZjODljYWI5ZjBhYmIxNzZcIl19In0
  * https://mandrillapp.com/track/click/30904755/www.blindsdirect.co.uk?p=eyJzIjoiWVZ5dm5GNWZIRm1DTUtvSEw3MlVjRFZ6aEJzIiwidiI6MSwicCI6IntcInVcIjozMDkwNDc1NSxcInZcIjoxLFwidXJsXCI6XCJodHRwczpcXFwvXFxcL3d3dy5ibGluZHNkaXJlY3QuY28udWtcXFwvZmFxcz91dG1fbWVkaXVtPWVtYWlsJnV0bV9zb3VyY2U9dHJhbnNhY3Rpb25hbCZ1dG1fY2FtcGFpZ249QmxpbmRzK0RpcmVjdCstK1NhbXBsZXMrMStkYXkrYXV0b21hdGlvblwiLFwiaWRcIjpcIjU0OTkwY2E2MjIxOTQ5NzQ5NzY0MjdkYWQ4NmIzZmM2XCIsXCJ1cmxfaWRzXCI6W1wiMjFiNzA3ZjcwN2I1MGY2MTJmN2IwMmQwYjllYzNlZTYwYWIwZDQxNFwiXX0ifQ
* Only *some* links in a [Cycling UK newsletter](https://a2a0a98917b84ecca6f4be18141f2010.marketingusercontent.com/m/view/r92u2fqE2xSxPwbNLYDc7ggQu1eZukxtGllx9pifJWcx#msdynttrid=d6nWs95y6tiA4F9042f2uh7EIrHsXpdl-XzgSc1KaqI):
  * The "Open this email in a browser" link doesn't work: https://a2a0a98917b84ecca6f4be18141f2010.svc.dynamics.com/t/t/g0AjucD1TNLRJZAuVAE0wAd1KXfRexZVPKeZrfeREdYx/r92u2fqE2xSxPwbNLYDc7ggQu1eZukxtGllx9pifJWcx?p0=r92u2fqE2xSxPwbNLYDc7ggQu1eZukxtGllx9pifJWcx
  * ...but all the other links work just fine, despite using the same `svc.dynamics.com` domain, e.g. the "write to your local candidates" link works: https://a2a0a98917b84ecca6f4be18141f2010.svc.dynamics.com/t/t/ygplKHDPGfhXlwShiwiFmPuWUXv24ltdMn8o5ctZ9E8x/r92u2fqE2xSxPwbNLYDc7ggQu1eZukxtGllx9pifJWcx
  * Even stranger, this message gets clipped by Gmail because it's quite long. If I then click the "View entire message" link at the bottom to open it in its own tab, the previously-broken "Open this email in a browser" link works just fine.

**Comment by ghostwords** [comment]
Hello and thanks for opening an issue!

Are you using Gmail in an Incognito window?

Probably related: #2998

**Comment by HarryCutts** [comment]
No, I'm using Gmail in my normal Chrome profile. I have it "installed" as a Progressive Web App (so that it gets its own window without tabs), but if I open gmail.com in a normal tab the issue still happens.

**Comment by ghostwords** [comment]
This looks like a Chrome Dev bug, not yet in Chrome Beta (127.0.6533.57).

**Comment by ghostwords** [comment]
Chromium bug: https://issues.chromium.org/issues/354748637#comment15

**Comment by HarryCutts** [comment]
Yep, this now seems to be fixed on Chrome 129.0.6630.0. Thanks for investigating!
